pennylane/qchem/convert.py

```diff
@@ -877,7 +898,14 @@ def _rccsd_state(ccsd_solver, tol=1e-15):
     # alpha, beta -> alpha, beta excitations
     rowvals, colvals = np.array(list(product(t1a_configs, t1b_configs)), dtype=int).T
```

Comment:
This looks like a third occurrence of similar code, i.e., building this dictionary form from method-specific configurations. We could clean up some code by making a separate helper function for doing this conversion all together which will help us condense repetitive code.

Reply:
We thought about this early on, but unfortunately there are enough small differences between the four occurrences you mention (RCISD/UCISD/RCCSD/UCCSD) that there is not a simple helper function to abstract this part away.

Reply:
I meant to condense these into a helper function - 
![image](https://github.com/PennyLaneAI/pennylane/assets/23234279/a9110cf2-baf5-4afb-9ee5-e1b71ba9ec96)

Reply:
I agree that it is possible to replace with a helper, but that helper will need 5-6 arguments and in my mind breaks up the flow / readability of the function. I prefer to keep the `einsum` explicit.